src/test/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngineUnitTest.java

```diff
@@ -89,4 +93,18 @@ public void testCalculateGenotypes() {
         final VariantContext vcOut1 = genotypingEngine.calculateGenotypes(vcSpanDel, GenotypeLikelihoodsCalculationModel.INDEL, null);
         Assert.assertFalse(vcOut1.getAlleles().contains(Allele.SPAN_DEL));
     }
+
+    @Test //test for https://github.com/broadinstitute/gatk/issues/2530
+    public void testNoIndexOutOfBoundsExceptionWhenSubsettingToNoAlleles(){
+        final VariantContext vc = new VariantContextBuilder(null, "1", 100, 100, Arrays.asList(refA, altT))
+                .genotypes(GenotypeBuilder.create(SAMPLES.getSample(0), Arrays.asList(refA, altT))).make();
```

Comment:
it works either way since the alt allele gets filtered out without any pl, but I've switch to a hom ref to be cleaer